storage/core/src/main/java/com/hortonworks/streamline/storage/StorageManager.java

```diff
@@ -68,6 +68,14 @@
     void addOrUpdate(Storable storable) throws StorageException;
 
     /**
+     * Updates an existing {@link Storable}. Throws {@link StorageException} if the
+     * storable could not be updated.
+     *
+     * @param storable the storable
+     */
+    void update(Storable storable);
```

Comment:
Raised: https://github.com/hortonworks/streamline/issues/846

>What happens if you try updating a storable that does not exist?

It will throw an exception.